src/libcore/slice.rs

```diff
@@ -785,6 +801,25 @@ impl<'a, T> Iter<'a, T> {
     pub fn as_slice(&self) -> &'a [T] {
         make_slice!(T => &'a [T]: self.ptr, self.end)
     }
+
+    // Helper function for Iter::nth
+    fn iter_nth(&mut self, n: usize) -> Option<&'a T> {
```

Comment:
It may be possible to share code between the iter_nth implementations but I don't know how safe transmuting/casting between const/mut pointers is...